manifests/interface.pp

```diff
@@ -172,32 +174,28 @@
     $peer = []
   }
 
-  systemd::network { "${interface}.netdev":
-    content         => epp("${module_name}/netdev.epp", {
-        'interface'   => $interface,
-        'dport'       => $dport,
-        'description' => $description,
-        'mtu'         => $mtu,
-        'peers'       => $peers + $peer,
-    }),
-    restart_service => true,
-    owner           => 'root',
-    group           => 'systemd-network',
-    mode            => '0440',
-    require         => File["/etc/wireguard/${interface}"],
-  }
-
-  $network_epp_params = {
-    'interface' => $interface,
-    'addresses' => $addresses,
-    'routes'    => $routes,
-  }
-
-  systemd::network { "${interface}.network":
-    content         => epp("${module_name}/network.epp", $network_epp_params),
-    restart_service => true,
-    owner           => 'root',
-    group           => 'systemd-network',
-    mode            => '0440',
+  case $provider {
+    'systemd': {
+      wireguard::provider::systemd { $interface :
+        interface   => $interface,
```

Comment:
If we change the provider of an interface (i.e. systemd -> wgquick), the old one will be kept in place and I suspect this will cause problems.  Maybe we can add an `ensure` parameters to `wireguard::provider::systemd` and `wireguard::provider::wgquick`, always instantiate both of them with an adequate `$ensure` parameter which is passed down to the actual resources?

Reply:
I added the ensure parameter, should we add if systemd is the provider that wgquick will be executed with ensure = absent and vice versa?